File: taskhive-backend/src/main/java/com/digiwork/taskhive/module/analytics/dto/EmployeePerformanceResponse.java
package com.digiwork.taskhive.module.analytics.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class EmployeePerformanceResponse {

    private UUID employeeId;
    private String employeeName;
    private int tasksAssigned;
    private int tasksCompleted;
    private BigDecimal onTimeRate;
    private BigDecimal avgCompletionHours;
}

File: taskhive_flutter/lib/features/analytics/data/models/employee_performance_response.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'employee_performance_response.freezed.dart';
part 'employee_performance_response.g.dart';

@freezed
class EmployeePerformanceResponse with _$EmployeePerformanceResponse {
  const factory EmployeePerformanceResponse({
    String? employeeId,
    String? employeeName,
    int? tasksAssigned,
    int? tasksCompleted,
    double? onTimeRate,
    double? avgCompletionHours,
  }) = _EmployeePerformanceResponse;

  factory EmployeePerformanceResponse.fromJson(Map<String, dynamic> json) =>
      _$EmployeePerformanceResponseFromJson(json);
}
